frontend/src/pages/facemash.js

```diff
@@ -30,75 +29,78 @@ class FaceMash extends Component<Props> {
   constructor(props) {
     super(props)
     this.state = {
-      results: this.props.result,
       error: false,
       message: ''
     }
   }
 
-  static async getInitialProps({ query }) {
-    // check whether query.id is real candidate
-    // TODO: do not fetch again, if user hasn't responded to previous match
-    try {
-      const { result } = await getCandidateMatch()
-      if (result === undefined) {
-        return { error: 'Bad Request' }
-      }
-      return { result }
-    } catch (err) {
-      console.log('Candidate Page error: ', err.message)
-      return { error: 'Bad Request' }
+  async getNewMatch() {
+    const { result } = await getCandidateMatch()
+    if (result === undefined) {
+      console.log('Could not get new FaceMash match')
+    }
+    console.log('Got new match')
+    const { generateMatchData } = this.props
+    generateMatchData(result.candidate1, result.candidate2, result.matchID)
+  }
+
+  componentDidMount() {
+    const { candidates } = this.props
+    if (candidates == null || candidates.length != 2) {
+      this.getNewMatch()
     }
   }
 
   handleClick = async e => {
     const winner = e.target.name
     try {
-      const { candidate1, candidate2, matchID } = this.state.results
+      const { candidates, matchID } = this.props
       const res = await setMatchWinner(
-        candidate1._id,
-        candidate2._id,
-        winner === '0' ? candidate1._id : candidate2._id,
+        candidates[0]._id,
+        candidates[1]._id,
+        winner === '0' ? candidates[0]._id : candidates[1]._id,
         matchID
       )
+      console.log('madeuit')
       if (res.success) {
         this.setState({
           success: 'Successfully Submitted'
         })
       }
-      // TODO: fetch for new match
+      this.getNewMatch()
     } catch (err) {
-      console.err(('Error': err.message))
+      console.error(('Error': err))
     }
   }
 
   render() {
     if (this.props.error) {
       return <div>Bad Fetch. Try again</div>
     }
-    return (
+    const { candidates } = this.props
+    return candidates && candidates.length == 2 ? (
       <div>
         <Head title="FaceMash" />
         <Nav />
         <Container>
           <p>{this.state.message}</p>
           <div className="row">
             <div className="col-md-6">
-              <Candidate candidate={this.state.results.candidate1} />
+              <Candidate candidate={candidates[0]} />
               <button name="0" className="btn btn-info" onClick={this.handleClick}>
                 Pick
               </button>
             </div>
             <div className="col-md-6">
-              <Candidate candidate={this.state.results.candidate2} />
+              <Candidate candidate={candidates[1]} />
               <button name="1" className="btn btn-info" onClick={this.handleClick}>
                 Pick
               </button>
             </div>
           </div>
         </Container>
       </div>
-    )
+    ) : null
```

Comment:
instead of returning null. return a place holder stating the error. like this https://github.com/hack4impact-uiuc/love-without-boundaries/blob/3c8ceb3d76c44ce15b8257cc147ae6cf3b092476/react_frontend/src/pages/StudentPage.js#L60

which would render something like this 
![image](https://user-images.githubusercontent.com/27740557/42799233-e7805b1e-894b-11e8-8605-ec52d5bde162.png)